doc/ignition.md

```diff
@@ -0,0 +1,107 @@
+
+The list of supported configuration providers are as follows:
+
+  Provider Name  |                        Description
+-----------------|-------------------------------------------------------------
+ cmdline         | Fetches the config from the URL provided via the
+                 : "coreos.config.url" kernel boot option.
+ file            | Read the config from a file named "config.json" in the
+                 : current working directory.
+
+## Configuration ##
+
+The Ignition configuration is provided in a JSON document via one of the
+aforementioned config providers. The format of the document is detailed in the
+following section.
+
+### Specification ###
+
+- **version** (integer): the version number of the spec. Must be `1`.
+- **storage** (object): describes the desired state of the systemd's storage
+                        devices.
+  - **disks** (list of objects): the list of disks to be configured and their
+                                 options.
+    - **device** (string): the absolute path to the device. Devices are
+                           typically referenced by the /dev/disk/by-* symlinks.
+    - **wipe-table** (boolean): when true, the partition tables are erased
+                                before any further manipulation. Otherwise, the
+                                existing entries are left in-tact.
+    - **partitions** (list of objects): the list of partitions and their
+                                        configuration for this particular disk.
+      - **label** (string): the PARTLABEL for the partition.
+      - **number** (integer): the partition number, which dictates it's
+                              position in the partition table.
+      - **size** (integer): the size of the partition (in sectors).
+      - **start** (integer): the start of the partition (in sectors).
+      - **type-guid** (string): the GPT [partition type GUID][part-types].
+  - **raid** (list of objects): the list of RAID arrays to be configured.
+    - **name** (string): ?
+    - **level** (string): the redundancy level of the array (e.g. linear,
+                          raid1, raid5, etc.).
+    - **devices** (list of strings): the list of devices (referenced by their
+                                     absolute path) in the array.
+    - **spares** (integer): the number of spares (if applicable) in the array.
+  - **filesystems** (list of objects): the list of filesystems to be
+                                       configured. Typically, one filesystem
+                                       is configured per partition.
+    - **device** (string): the absolute path to the device. Devices are
+                           typically referenced by the /dev/disk/by-* symlinks.
+    - **initialize** (boolean): when true, any existing existing data on the
+                                partition is destroyed during the formatting.
+                                Otherwise, if an existing filesystem is
```

Comment:
this isn't how it's implemented today.

If true, we initialize the filesystem.  If false, we don't do any initialization here but will mount an existing filesystem if you've included files destined for this filesystem.